### PR TITLE
fixes and new provider

### PR DESCRIPTION
**Pull request by Stormunblessed**
- Theflix.to Fixed
- Homepage fixed for Monoschinos
- And added VidSrc

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt [hunk 1]
```diff
@@ -92,9 +94,37 @@ class TheFlixToProvider : MainAPI() {
...
+                "Content-Type" to "application/json;charset=utf-8",
+                "Content-Length" to "35",
+                "Origin" to "https://theflix.to",
+                "DNT" to "1",
+                "Connection" to "keep-alive",
+                "Referer" to "https://theflix.to/",
+                "Sec-Fetch-Dest" to "empty",
+                "Sec-Fetch-Mode" to "cors",
+                "Sec-Fetch-Site" to "same-site",)
+        ).okhttpResponse.headers.values("Set-Cookie")
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt:118]: you should be able to just do .cookies and it does all the parsing for you

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt [hunk 2]
```diff
@@ -92,9 +94,37 @@ class TheFlixToProvider : MainAPI() {
         @JsonProperty("conversionDate") val conversionDate: String? = null,
         @JsonProperty("id") val id: Int? = null,
         @JsonProperty("available") val available: Boolean? = null,
+        @JsonProperty("videos"           ) var videos           : ArrayList<String>? = arrayListOf(),
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt:97]: val

Comment (LagradOst) [reply app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt:97]: and val on every other JsonProperty

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt [hunk 3]
```diff
@@ -92,9 +94,37 @@ class TheFlixToProvider : MainAPI() {
...
     )
 
 
+    private suspend fun getCookies(): Map<String, String> {
+        val cookieResponse = app.post(
+            "https://theflix.to:5679/authorization/session/continue?contentUsageType=Viewing",
+            headers = mapOf(
+                "Host" to "theflix.to:5679",
+                "User-Agent" to "Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0",
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt:106]: USER_AGENT

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt [hunk 4]
```diff
@@ -92,9 +94,37 @@ class TheFlixToProvider : MainAPI() {
...
+                "Accept" to "application/json, text/plain, */*",
+                "Accept-Language" to "en-US,en;q=0.5",
+                "Content-Type" to "application/json;charset=utf-8",
+                "Content-Length" to "35",
+                "Origin" to "https://theflix.to",
+                "DNT" to "1",
+                "Connection" to "keep-alive",
+                "Referer" to "https://theflix.to/",
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt:114]: $mainUrl

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt [hunk 5]
```diff
@@ -92,9 +94,37 @@ class TheFlixToProvider : MainAPI() {
         @JsonProperty("conversionDate") val conversionDate: String? = null,
         @JsonProperty("id") val id: Int? = null,
         @JsonProperty("available") val available: Boolean? = null,
+        @JsonProperty("videos"           ) var videos           : ArrayList<String>? = arrayListOf(),
     )
 
 
+    private suspend fun getCookies(): Map<String, String> {
+        val cookieResponse = app.post(
+            "https://theflix.to:5679/authorization/session/continue?contentUsageType=Viewing",
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/TheFlixToProvider.kt:103]: $mainUrl

**Review by LagradOst — COMMENTED** [review]
Minor code quality changes 👍

**Comment by LagradOst** [comment]
LGTM, ready to merge? 👍

**Comment by Stormunblessed** [comment]
> LGTM, ready to merge? 👍 

Yed

**Pull request merged**